### DISCUSSION TITLE
Welcome to cppcms Discussions!

### DISCUSSION THREAD

**Discussion by artyom-beilis** [artyom-beilis/cppcms#94] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by masaoliou** [comment]
### [How Hertz Paid Accenture $32 Million for a Website That Never Went Live](https://www.henricodolfing.com/2019/10/case-study-hertz-accenture-website.html)

**What Went Wrong**:

1. **Front End Layer** – The presentation layer in which the screens and the interface were presented to the users using the JavaScript framework Angular.
2. **Content Management** – Adobe Experience Manager (AEM) was the tool of choice to update and revise the content that appears on the website.
3. **Microservices Layer** – Composed of code that performed various “services,” such as searching for a Hertz location or for a type of vehicle, writing a review, sending an email to a Hertz representative, etc.
4. **Integration Layer** – MuleSoft was the tool of choice to allow the front-end systems to request and receive data from the back-end systems.
5. **Back-End Systems** – Hertz’s reservation systems, rewards systems, etc.

results:

- **No responsive design**
- **No common core**
- **Vulnerable code**
- **No experience with used technologies**
- **No testing and documentation**

Hertz filed bankruptcy on 2021-07-21.

### They should have used _only CppCMS_ to implement _layers 2., 3., 4._ and perhaps also _5._.

**Comment by artyom-beilis** [comment]
CppCMS is very good for several things:

1. High performance on low end hardware, especially if you do some heavy C++ computation logic
2. Embedded systems
3. Implementing web services for existing C++ system or a system that requires running in C++ due to performance or other constrains.

I wouldn't take it for some generic web development because - experienced C++ software engineer costs much more that typical fulls stack, it is easier to find software developers for angular/nodejs than for CppCMS and of course more existing tools.

Bottom line... Also I'm developer of CppCMS it isn't an universal tool to do everything.

**Comment by masaoliou** [comment]
It's not easy to become a senior C++ programmer, but I don't think it's easy for programmers to build a decent website using SOA, either.

Website developers want to decrease the number of components or "layers" when they implement back end servers, not to increase them (and accordingly the number of points of failure).

I have not yet encountered any website I can not build its back end server using only the following elements:

- CppCMS
- library curl
- BOOST upgradable lock
- PostgreSQL
- Lighttpd (as Artyom seemed to recommend)

Perhaps someday people will eventually abandon new fashioned website development frameworks or languages and come back to embrace the good old CppCMS whose reliability is bullet proof, and [its throughput is allegedly unbeatable](http://cppcms.com/wikipp/en/page/benchmarks_all).

Here is the bonus with CppCMS - The source codes of all my websites are incredibly tiny, meaning they are easy to maintain.
